File: algorithms/L/Linked List in Binary Tree/Linked List in Binary Tree.py
class Solution(object):
    def isSubPath(self, head, root):
        if not root:    
            return False
        if self.issame(head, root):
            return True
        return self.isSubPath(head, root.left) or self.isSubPath(head, root.right)
    def issame(self, head, root):
        if not head:
            return True
        if not root:
            return False
        if head.val != root.val:
            return False
        return self.issame(head.next, root.left) or self.issame(head.next, root.right)

File: algorithms/L/Linked List in Binary Tree/Linked List in Binary Tree.java
class Solution {
    public boolean isSubPath(ListNode head, TreeNode root) {
        if(root == null)    return false;
        if(issame(head, root))  return true;
        return isSubPath(head, root.left) || isSubPath(head, root.right);
    }
    private boolean issame(ListNode head, TreeNode root) {
        if(head == null)    return true;
        if(root == null)    return false;
        if(head.val != root.val)    return false;
        return issame(head.next, root.left) || issame(head.next, root.right);
    }
}
